### ISSUE TITLE
Do some proper benchmarking

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/gateplugin-Format_Bdoc#1]
Preliminary result on tac2012-fix02 corpus, processed with ANNIE, loading/saving single-threaded in the GUI, using 20G RAM, SSD file system:

Comparison with gatexml=1:
* finf: saving 1.8 slower, loading 1.18 faster, size 5.5 smaller
* bdocjson: saving 55 faster, loading 13 faster, size 3.6 smaller
* bdocjson/gzip: saving 24 faster, loading 9.1 faster, size 31.9 smaller

e.g. saving to gatexml takes 2:44, creates 1.4G, saving to bdocjson takes 0:02.5, creates 394M

**Comment by johann-petrak** [comment]
For comparison: on the same system using a simple python script, we need:
* for loading bdocjson: 47 seconds (Java 7 seconds)
* for saving bdocjson:  59 seconds (Java 3 seconds)

**Comment by johann-petrak** [comment]
Running the gs8:bench-gcp benchmarks:

All runs with -t 20 -m 20G

Input only, no output:
```
finf:      TIME: elapsed=793.46 cpu=455%   (5.5 * gatexml)
gatexml    TIME: elapsed=143.57 cpu=1727%   (=1)                     
bdocjson:  TIME: elapsed=43.26 cpu=1745%  (gatexml / 3.32)
bdocjsongz:TIME: elapsed=113.04 cpu=630%  (gatexml / 1.27)                      

GCP, input is bdocjson output to:
gatexml:  TIME: elapsed=63.56 cpu=1820% (=1)
finf:     TIME: elapsed=63.32 cpu=1843% (1 * gatexml)
bdocjson: TIME: elapsed=53.39 cpu=1880% (gatexml / 1.2) 
```                                                                               

Space:
```
gatexml:       32G
finf:                 6G  (gatexml / 5.3)
bdocjson:       8.5G  (gatexml / 3.8)
bdocjsongz:   742M (gatexml / 44.2)
```

**Comment by greenwoodma** [comment]
Are you sure these numbers are right? I'm only asking as the finf numbers look very odd. My original testing (see http://englishjavadrinker.blogspot.com/2013/08/savings-in-time-and-space.html) showed that finf was faster to both load and save documents than GATE XML. These numbers though seem to suggest finf is 5.5 times slower to load and basically the same to save as GATE XML. I know the testing setup is different but it seems odd they should be so different.

**Comment by johann-petrak** [comment]
Yes, I was surprised as well, especially since the first benchmark, with a different corpus and just the Gate GUI had finf loading (a bit) faster than gatexml. the results look very different with the gcp benchmark. Oddly, CPU percentage is very different between finf and gatexml as well.

Never sure there was no mistake, so I am happy to see others replicate or do their own benchmarks.
I can send you the details of what I did where, if you are interested.

**Comment by greenwoodma** [comment]
Yes, I'd be interested in the details of how you ran it. I'm sure you're right and that the numbers in GCP are just different, but I'd like to have a dig around to see why (in case I've messed something up in the finf GCP handler)

**Comment by johann-petrak** [comment]
Continue this later when https://github.com/GateNLP/gate-core/issues/122 got fixed.
